### PR TITLE
Bdd features and scenarios

### PR DESCRIPTION
**Pull request by bocharsky-bw**
Resolve `02-bdd-features` and `03-scenarios` in https://github.com/knpuniversity/behat/issues/12

### CODE REVIEW

**Comment by bocharsky-bw** [comment]
Do you have `03-scenarios` challenges already?

File: knpu/Challenges/BddFeatures/RegistrationFeatureCoding.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        return self::EXECUTION_MODE_PHP_NORMAL;
+    }
+
+    public function setupContext(CodingContext $context)
+    {
+    }
+
+    public function grade(CodingExecutionResult $result)
+    {
+        throw new GradingException('Oops...');
```
Comment (bocharsky-bw) [inline knpu/Challenges/BddFeatures/RegistrationFeatureCoding.php]: @weaverryan I wondering how to grading features. There're any best solution for it?

Comment (weaverryan) [inline knpu/Challenges/BddFeatures/RegistrationFeatureCoding.php]: I was thinking we do some basic parsing to make sure their format is correct - I.e. they have "in order to", "as a", etc. there's a new gherkin execution mode I added to activity-runner, which basically renders no output (it doesn't have anything to help with parsing gherkin though)

Comment (bocharsky-bw) [inline knpu/Challenges/BddFeatures/RegistrationFeatureCoding.php]: Should we also to chech the correct order of "in order to", "as a", etc.?

Comment (weaverryan) [inline knpu/Challenges/BddFeatures/RegistrationFeatureCoding.php]: I think so - e.g line 1 should be Feature:, line 2 In order to, etc

**Comment by weaverryan** [comment]
@bocharsky-bw I finished everything through 03. Sorry about not having the challenges ready for 03 - I *thought* I had added them, but I did not. I'll add more challenges soon for the other chapters.

But hey... first chapters are now live :)

**Comment by bocharsky-bw** [comment]
@weaverryan OK, I could still work with code blocks now.

And now I have an illustrative example how to grade challenges :)

BTW, we could use RegEx to grade more complex things in the future.

**Comment by weaverryan** [comment]
I'm fine with Regex, I'm not great at it personally, and sometimes it can make things less readable (because you really need to understand regex to understand what the code is doing). But in general, I'm definitely cool with using regex when it makes sense.

And yes - code blocks would be awesome for now :)

**Pull request merged**